src/custom/state/orders/updaters/PendingOrdersUpdater.ts

```diff
@@ -13,6 +19,24 @@ import { OrderID } from 'api/gnosisProtocol'
 
 import { fetchOrderPopupData, OrderLogPopupMixData } from 'state/orders/updaters/utils'
 
+/**
+ * Return the ids od the orders that we are not yet aware that are signed.
+ * This is, pre-sign orders, in state of "PRESIGNATURE_PENDING", for which we now know they are signed
+ *
+ * Used as an auxiliar method to detect which orders we should mark as pre-signed, so we change their state
+ *
+ * @param allPendingOrders All pending orders
+ * @param signedOrdersIds ids of orders we know are already pre-signed
+ * @returns ids of the pending orders that were pending for pre-sign, and we now know are pre-signed
+ */
+function _getPresignedOrdersStillWaiting(allPendingOrders: Order[], signedOrdersIds: OrderID[]) {
+  const presignaturePendingIds = allPendingOrders
+    .filter((order) => order.status === OrderStatus.PRESIGNATURE_PENDING)
```

Comment:
You can combine both conditions in the same filter:

```suggestion
    .filter((order) => order.status === OrderStatus.PRESIGNATURE_PENDING && presignaturePendingIds.includes(order.id))
```

Reply:
Am i missing something, or u try to use `presignaturePendingIds` before time? 

First filter, just selects the orders for which we are still waiting to be signed.

Reply:
Yeah, I used the wrong one.
I meant to use `signedOrderIds` instead of `presugnaturePendingIds`

Reply:
Good! i was doing in two steps since it looked easier to read to me, but your proposal is fine. note that these lists are not too big, so imo there's not too much of a difference.